### PR TITLE
[Shader Graph] Fix 5 HUP Bugs for Sub Graphs

### PR DESCRIPTION
**Pull request by alindmanUnity**
### Purpose of this PR
Why is this PR needed, what hard problem is it solving/fixing?
Fix 5 HUP bugs related to sub graphs.
1) `1183805 `- Saving a sub graph in HDRP throws null reference.
- Saving sub graphs no longer cause a null reference error. 
2) `1183808` +` 1183815` + `1183819` - Deleted or missing sub graph assets throw unclear errors and null refs. 
- If a sub graph or shader graph references a missing or deleted sub graph asset, errors are now clear and the editor no longer locks up. 
3) `1183782` - Sub graph assets can reference themselves via drag and drop 
- Sub graph assets now check against their own guid, drag and drop no longer allows recursive behavior. 

---
### Release Notes
Please add any useful notes about the feature/fix that might be helpful for others.
Just a couple of checks for null assets. 

---
### Testing status
**Katana Tests**: First off we need to make sure the Katana SRP tests are green?

**Manual Tests**: What did you do?
- [x] Opened test project + Run graphic tests locally
- [x] C# and shader warnings (supress shader cache to see them)
- [x] Checked new resources path for the reloader (in devloper mode, you have a button at end of resources that check the pathes)
- Other: 

**Automated Tests**: What did you setup? (Add a screenshot or the reference image of the test please)
Alternative, launch Yamato (Select your branch):
https://yamato.prd.cds.internal.unity3d.com/jobs/78-ScriptableRenderPipeline/tree/sg%252Ffix-hup-subgraph

Any test projects to go with this to help reviewers?

---
### Overall Product Risks
**Technical Risk**: None, Low, Medium, High? Low

**Halo Effect**: None, Low, Medium, High? Low

---
### Comments to reviewers
Notes for the reviewers you have assigned.

### CODE REVIEW

File: com.unity.shadergraph/CHANGELOG.md [hunk 1]
```diff
@@ -29,6 +29,8 @@ and this project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.
 - PBR master nodes now calculate Global Illumination (GI) correctly.
 - PBR master nodes now apply surface normals.
 - PBR master nodes now apply fog.
+- Missing or deleted Sub Graph assets now display correct errors.
```
Comment (sharlenet) [inline com.unity.shadergraph/CHANGELOG.md]: I feel like this should be:
The Editor now displays correct errors for missing or deleted Sub Graph Assets.
or
Shader Graph now displays correct errors for missing or deleted Sub Graph Assets.

File: com.unity.shadergraph/CHANGELOG.md [hunk 2]
```diff
@@ -29,6 +29,8 @@ and this project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.
...
+- Sub Graph assets can no longer accept recursive nodes via drag and drop.
```
Comment (sharlenet) [inline com.unity.shadergraph/CHANGELOG.md]: Does this work?
You can no longer drag and drop recursive nodes onto Sub Graph Assets.

Comment (alindmanUnity) [reply com.unity.shadergraph/CHANGELOG.md]: yes! that's correct

File: com.unity.shadergraph/CHANGELOG.md (hunk 1 above)
Comment (alindmanUnity) [reply com.unity.shadergraph/CHANGELOG.md]: The first one is more accurate, it's logs in the console in addition to flags in the graph.

**Pull request merged**